### PR TITLE
web: Add more browsers to the browser testing, including mobile browsers

### PR DESCRIPTION
**Pull request by Dinnerbone**
This separates `npm run test` into `npm run test` (regular node tests) and `npm run wdio` (browser based tests, using wdio).

Documentation has been adjusted, and `wdio` no longer just assumes Chrome is installed; you now have to tell it which browsers to test on. This also means it now supports more than just Chrome locally.

In addition, there's a `--browserstack` option for running the tests on Browserstack. This is **not done by CI yet**, as running privileged code in Pull Request actions on Github is a whole problem in itself. At the very least, being able to do this locally is good enough for now!

### CODE REVIEW

File: web/README.md [hunk 1]
```diff
@@ -112,9 +112,38 @@ run a demo locally with `npm run demo`, or [install the extension in your browse
 
 ### Testing
 
-To run all of the tests in this project, we currently require that you have [Chrome installed to its default location](https://www.google.com/chrome/).
+There are two parts of tests to this project:
+- Regular node tests, ran through `npm run test`. You must have built everything first as above. These have no special requirements.
+- Browser based tests, ran through `npm run wdio` with extra arguments as below. These take longer to run and require some setup.
 
-First, ensure you've build every package (see above), and then run `npm run test` to run the full suite of tests.
+## Browser based tests
```
Comment (torokati44) [inline web/README.md:119]: nit: Section name not in title case.

File: web/README.md [hunk 2]
```diff
@@ -112,9 +112,38 @@ run a demo locally with `npm run demo`, or [install the extension in your browse
...
+- `--chrome` for Chrome
+- `--firefox` for Firefox
+- `--edge` for Edge
+
+### Browserstack (Mobile browsers)
+To run tests on mobile devices on Browserstack, pass the `--browserstack` argument.
+We also have our "minimum supported desktop browsers" available too, by additionally passing `--oldVersions`.
+
+You will need a Browserstack account (Maintainers may contact @Dinnerbone on Discord for an invite to the Ruffle team),
+and set the appropriate values to `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` environment variables.
```
Comment (torokati44) [inline web/README.md:136]: nits: Section names not in title case, and I think they call themselves "BrowserStack".

File: web/README.md [hunk 3]
```diff
@@ -112,9 +112,38 @@ run a demo locally with `npm run demo`, or [install the extension in your browse
...
+- Regular node tests, ran through `npm run test`. You must have built everything first as above. These have no special requirements.
+- Browser based tests, ran through `npm run wdio` with extra arguments as below. These take longer to run and require some setup.
 
-First, ensure you've build every package (see above), and then run `npm run test` to run the full suite of tests.
+## Browser based tests
+There are full integration tests that require a browser to run. We don't make any assumptions about your environment, and so you must specify it yourself.
+
+To run these tests, first build the project as above, then use `npm run wdio -- --arg1 --arg2` etc.
+
+### Local Browsers
```
Comment (torokati44) [inline web/README.md:124]: Okay, maybe this is the exception for being in title case...? Either way, it's inconsistent.

**Comment by torokati44** [comment]
_Ladybird and Servo when?_ :eyes:

**Comment by Dinnerbone** [comment]
> _Ladybird and Servo when?_ :eyes:

When they make and publish webdrivers!

**Comment by Dinnerbone** [comment]
Fixed all!

**Pull request merged**